### PR TITLE
support observe cache usage and hit status

### PR DESCRIPTION
**Pull request by xzxiong**
## What type of PR is this?

- [ ] API-change
- [ ] BUG
- [ ] Improvement
- [ ] Documentation
- [x] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue # https://github.com/matrixorigin/matrixone/issues/19416
https://github.com/matrixorigin/matrixone/issues/19415

## What this PR does / why we need it:
1.  refactor Cache postSet, postGet, postEvict function as `func(..., size int64)`
2. metric: show all cache: 1) hit, read counter; 2) inuse, capacity bytes.
    - including: fs mem cache, fs disk cache, fs metadata cache.
3. dashboard: add 2 part to show: 1) cache hit; 2) cache usage.
4. query_service: add rpc api `GoGCPercent` to call `debug.SetGCPercent`
5. query_service: fix `mem` cache register while NewMemCache with the name. -- for Evict.
6. query_service: log the calling history.

dashboard preview:
image

### CODE REVIEW

File: pkg/fileservice/mem_cache.go [hunk 1]
```diff
@@ -65,20 +85,18 @@ func NewMemCache(
...
-	}
 	dataCache := fifocache.NewDataCache(capacityFunc, postSetFn, postGetFn, postEvictFn)
 
-	return &MemCache{
+	ret := &MemCache{
 		cache:       dataCache,
 		counterSets: counterSets,
 	}
+
+	if name != "" {
```
Comment (xzxiong) [inline pkg/fileservice/mem_cache.go:95]: fix for `handleFileServiceCacheEvictRequest`, @reusee  ptal

File: pkg/objectio/cache.go [hunk 2]
```diff
@@ -109,23 +111,32 @@ func cacheCapacityFunc(size int64) fscache.CapacityFunc {
...
-		metaCache = fifocache.New[mataCacheKey, []byte](
-			cacheCapacityFunc(size),
-			shardMetaCacheKey,
-			nil, nil, nil,
-		)
+		metaCache = newMetaCache(cacheCapacityFunc(size))
 	})
 }
 
+func newMetaCache(capacity fscache.CapacityFunc) *fifocache.Cache[mataCacheKey, []byte] {
```
Comment (xzxiong) [inline pkg/objectio/cache.go:123]: refactor to add `postSet`, `postEvict` function

File: pkg/defines/const.go [hunk 3]
```diff
@@ -46,6 +46,7 @@ const (
 	MORPCMinVersion    int64 = math.MinInt64
 	MORPCVersion1      int64 = 1
 	MORPCVersion2      int64 = 2
-	MORPCVersion3      int64 = 3 // start from 1.3.0, 1.2.3
-	MORPCLatestVersion       = MORPCVersion3
+	MORPCVersion3      int64 = 3 // start from 1.3.0
+	MORPCVersion4      int64 = 4 // start from 2.0.1
```
Comment (xzxiong) [inline pkg/defines/const.go:50]: new version for
- CmdMethod_MetadataCache
- CmdMethod_GOGCPercent

File: pkg/fileservice/mem_cache.go (hunk 1 above)
Comment (reusee) [reply pkg/fileservice/mem_cache.go:91]: LGTM

File: pkg/fileservice/fifocache/fifo.go [hunk 4]
```diff
@@ -306,3 +307,7 @@ func (c *Cache[K, V]) Evict(ctx context.Context, done chan int64) {
 	}
 	c.evict(ctx, done, 0)
 }
+
+func (c *Cache[K, V]) Capacity() int64 {
```
Comment (reusee) [inline pkg/fileservice/fifocache/fifo.go:311]: 没有用到的话，就不需要加新的方法

**Review by reusee — CHANGES_REQUESTED** [review]

**Review by zhangxu19830126 — APPROVED** [review]

File: pkg/fileservice/fifocache/fifo.go (hunk 4 above)
Comment (xzxiong) [reply pkg/fileservice/fifocache/fifo.go]: got it.

**Review by reusee — APPROVED** [review]

**Review by LeftHandCold — APPROVED** [review]

**Review by daviszhen — APPROVED** [review]

**Review by sukki37 — APPROVED** [review]

**Pull request merged**